daemon/daemon_unix.go

```diff
@@ -265,8 +269,8 @@ func verifyContainerResources(resources *containertypes.Resources) ([]string, er
 		warnings = append(warnings, "You specified a kernel memory limit on a kernel older than 4.0. Kernel memory limits are experimental on older kernels, it won't work as expected and can cause your system to be unstable.")
 		logrus.Warnf("You specified a kernel memory limit on a kernel older than 4.0. Kernel memory limits are experimental on older kernels, it won't work as expected and can cause your system to be unstable.")
 	}
-	if resources.OomKillDisable && !sysInfo.OomKillDisable {
-		resources.OomKillDisable = false
+	if resources.OomKillDisable != nil && !sysInfo.OomKillDisable {
+		resources.OomKillDisable = nil
```

Comment:
It's not about how many kinds of value can be set, it's about if the default value has specific meaning. Say if you set `OomKillDisable` as true when creating the container, then you want to update memory and you send only `Memory` in `Resources`, but since we decode the whole Resources struct, we'll get `OomKillDisable=false`, which will be updated unexpected.